Feat login

dani-santos-code opened the pull request:
This implements the Login/Logout, following the requirements in our wireframe: https://www.figma.com/file/GsVzEmXeqvf8MsoaAOlMB5D9/CODE-Resource-Center?node-id=0%3A1

**Important**: According to our wireframe, a user should be authenticated with an email. But the Django auth module enforces authentication with the username, which is the unique element in the model. I'm investigating the issue and will try out a couple of things today. But for now, let's leave it like this. I believe the most important thing about this PR is the routing and the conditional rendering of options on the Navbar.

**UPDATE**: According to the documentation it's very difficult to customize after we've already started the project and run the migrations: https://docs.djangoproject.com/en/1.11/topics/auth/customizing/#auth-custom-user

(for testing, use these)
**user:** crcuser / **password :** myresources

If user is not logged in he/she cannot access any other of our paths and will always bounce back to the Login page. 

Here are some gifs to illustrate the behavior. Please check out this branch to see the functionality.

**Flow 1**  (user logs in with right password, then logs out):

![2018-11-17 12 09 00](https://user-images.githubusercontent.com/22174780/48660436-aa054700-ea61-11e8-9f51-eb544ae98116.gif)

**Flow 2** (user has the wrong password / username):
![2018-11-17 12 03 31](https://user-images.githubusercontent.com/22174780/48660371-db314780-ea60-11e8-981a-d966e7235885.gif)

**Flow 3** (user needs to register / signup):
![2018-11-17 12 04 53](https://user-images.githubusercontent.com/22174780/48660383-061b9b80-ea61-11e8-90dc-1f93be2e2be2.gif)

**Flow 4** (user is not authenticated and tries to access the dashboard):

![2018-11-17 12 05 35](https://user-images.githubusercontent.com/22174780/48660396-221f3d00-ea61-11e8-8924-c3621c9f2c3c.gif)

**Notice**: the Navbar for unauthorized users only has *Sign Up* and *Login*. But for authorized users, for now, you'll see Add Resource and Log out. We should also add settings once we have specified that task.  Whoever picks those tasks can add more options to our Navbar later! :) 

Thanks for your feedback! 🙏

gabriela99 commented on Makefile:
```diff
@@ -0,0 +1,14 @@
+	python manage.py makemigrations
+
+help:
+	@echo "\x1b[32;01m"make server "\x1b[30;01m" '-->' to start the server
```
Cool colours!

gabriela99 commented on README.md:
```diff
@@ -38,3 +38,7 @@ To run the admin panel and make sure the DB is working fine, go to `crc project`
+
+## Make files
+
+To make life easier, run commands with `make` on your shell. For a list of commands, run `make help`. 
```
I ran into problems because of Mojave, but running `xcode-select --install` fixed it

dani-santos-code replied:
I added it! :)

dani-santos-code replied:
@gabriela99, please check it out!

Commit:
Add pycache changes.

Commit:
Merging (pycache)

Commit:
Removed pycache from local

Commit:
added changes from development branch

Commit:
Add login layout and functionality.
- Paths were guarded, so if the user is not logged in, he/she cannot access the dashboard.
Default page is the Login. If user is logged in he/she can access Dashboard as well as Log out

Commit:
made sure that unauthorized users won't acccess our dashboard. So if user is not authenticated he/she will always bounce back to login page

Commit:
fixes UX issues on base html file

Commit:
centered H1 tag

gabriela99 reviewed (approved):
I seem to not be able to log in with crcuser, but it's fine with my own details. Footer and the body look great. I think we should use a different navbar for sign in since "sign up" and "logout" aren't relevant here. Besides that, looks and works great!
